src/rfbzero/redox_flow_cell.py

```diff
@@ -413,37 +413,48 @@ def _coulomb_counter(
+        self.prev_c_ox_ncls = self.c_ox_ncls
+        self.prev_c_red_ncls = self.c_red_ncls
+
         # update CLS and NCLS concentrations
-        c_ox_cls = self.c_ox_cls - delta_cls
-        c_red_cls = self.c_red_cls + delta_cls
-        c_ox_ncls = self.c_ox_ncls + delta_ncls
-        c_red_ncls = self.c_red_ncls - delta_ncls
+        new_c_ox_cls = self.c_ox_cls - delta_cls
+        new_c_red_cls = self.c_red_cls + delta_cls
+        new_c_ox_ncls = self.c_ox_ncls + delta_ncls
+        new_c_red_ncls = self.c_red_ncls - delta_ncls
 
         # for no crossover situation
         delta_ox_mols = 0.0
         delta_red_mols = 0.0
 
         # Coulomb counting from optional degradation and/or crossover mechanisms
         if cls_degradation is not None:
-            c_ox_cls, c_red_cls = cls_degradation.degrade(c_ox_cls, c_red_cls, self.time_step)
+            delta_ox_cls, delta_red_cls = cls_degradation.degrade(self.c_ox_cls, self.c_red_cls,
+                                                                  self.time_step)
+            new_c_ox_cls += delta_ox_cls
+            new_c_red_cls += delta_red_cls
 
         if ncls_degradation is not None:
-            c_ox_ncls, c_red_ncls = ncls_degradation.degrade(c_ox_ncls, c_red_ncls, self.time_step)
+            delta_ox_ncls, delta_red_ncls = ncls_degradation.degrade(self.c_ox_ncls, self.c_red_ncls,
+                                                                     self.time_step)
+            new_c_ox_ncls += delta_ox_ncls
+            new_c_red_ncls += delta_red_ncls
 
         if cross_over is not None:
-            (c_ox_cls, c_red_cls, c_ox_ncls, c_red_ncls, delta_ox_mols,
-             delta_red_mols) = cross_over.crossover(self.geometric_area, c_ox_cls, c_red_cls, c_ox_ncls, c_red_ncls,
-                                                    self.volume_cls, self.volume_ncls, self.time_step)
-        # update concentrations to self
-        self.prev_c_ox_cls = self.c_ox_cls
-        self.prev_c_red_cls = self.c_red_cls
-        self.prev_c_ox_ncls = self.c_ox_ncls
-        self.prev_c_red_ncls = self.c_red_ncls
-
-        self.c_ox_cls = c_ox_cls
-        self.c_red_cls = c_red_cls
-        self.c_ox_ncls = c_ox_ncls
-        self.c_red_ncls = c_red_ncls
+            delta_ox_cls, delta_red_cls, delta_ox_ncls, delta_red_ncls, delta_ox_mols, delta_red_mols = \
+                cross_over.crossover(self.geometric_area, self.c_ox_cls, self.c_red_cls, self.c_ox_ncls, self.c_red_ncls,
+                                     self.volume_cls, self.volume_ncls, self.time_step)
+            new_c_ox_cls += delta_ox_cls
+            new_c_red_cls += delta_red_cls
+            new_c_ox_ncls += delta_ox_ncls
+            new_c_red_ncls += delta_red_ncls
+
+        # Update new concentrations to self
+        self.c_ox_cls = new_c_ox_cls
+        self.c_red_cls = new_c_red_cls
+        self.c_ox_ncls = new_c_ox_ncls
+        self.c_red_ncls = new_c_red_ncls
```

Comment:
i think crossover should be treated differently than degradation because crossover requires concentrations as inputs (c_ox_cls, c_red_cls, c_ox_ncls, c_red_ncls), not delta_ox_cls... because it has to compare absolute concentration values.